### PR TITLE
Update LLVM's license to Apache 2.0

### PR DESCRIPTION
**Pull request by bradcray**
While reviewing some of our third-party licenses, I happened to notice that the copy of LLVM we redistribute is actually Apache 2.0 licensed, meaning our file was out of date.

The last release with UIUC license was LLVM 8 although some later releases have legacy licenses with UIUC.

The license was changed to Apache License v2.0 with LLVM Exceptions in llvm-project commit 469bdefd448b76c5adcdd67256e9a44fabf7e027 in Jan 2019.

Docs https://llvm.org/docs/DeveloperPolicy.html#legacy indicate some of LLVM still uses the old license but as far as we know, nothing that Chapel uses falls into that category.

### CODE REVIEW

**Comment by bradcray** [comment]
@jabraham17 / @mppf : Tagging you as an FYI and thanks for your help.  I don't think this actually needs a review, though.

**Review by mppf — APPROVED** [review]
Could you summarize Jade/my comments a bit more tightly before merging? Thanks.

Also if you want to be completely accurate, you could call the license "Apache 2.0 with LLVM Exceptions".

**Comment by mppf** [comment]
Here is my recommended PR message wording.

While reviewing some of our third-party licenses, I happened to notice that the copy of LLVM we redistribute is actually Apache 2.0 licensed, meaning our file was out of date.

The last release with UIUC license was LLVM 8 although some later releases have legacy licenses with UIUC.

The license was changed to Apache License v2.0 with LLVM Exceptions in llvm-project commit 469bdefd448b76c5adcdd67256e9a44fabf7e027 in Jan 2019.

Docs https://llvm.org/docs/DeveloperPolicy.html#legacy indicate some LLVM stuff still uses the old license but as far as we know, nothing that Chapel uses falls into that category.
